### PR TITLE
Remove AxisInfo from getTickProps

### PR DESCRIPTION
**Pull request by loichuder**
<s>
## Motivation
Getting back to the integer ticks problems, I felt that the current implementation was sometimes hard to follow:
- `AxisInfo` and `AxisConfig` are quite similar but used for different purposes
- The distinction between `IndexAxisConfig` and `DataAxisConfig` leads to additional code (different types, a type guard) as both are again very similar. Also, it is not clear from first approach that `IndexAxisConfig` forces the axis to contain only integers.

Reading the code, I found myself sometimes misdirected by all these different but similar types.

## Refactor
To address these points, I did the following:
- I removed `AxisInfo`. `AxisConfig` is used instead with a new boolean member `onlyIntegers`.
- The boolean `onlyIntegers` makes the distinction between `IndexAxisConfig` and `DataAxisConfig` superfluous. Both are merged in `AxisConfig` by changing `dataDomain` and `indexDomain` to `domain`.
- Rather than passing the whole config to `getTicksProp`, only `onlyIntegers` is passed as arg.

## Aftermath
- I hope it makes the code clearer as we don't have anymore to compare `AxisInfo` and `AxisConfig` to show what does what. The same object is used from start to finish.
- The introduction of the boolean `onlyIntegers` should also render the config more readable and easily usable.
- See other comments below
</s>

In the end, the refactor didn't feel right:
 - To keep the API safe (that is forcing linear scale for index/integer axis), I ended up needing to mutate `AxisConfig`, which conceptually, was the purpose of `AxisInfo`... 
- Also, for the same reason, `IndexAxisConfig` and `DataAxisConfig` typings could not be easily merged as `scaleType` serves no purpose when `isIndexAxis` is `true`.

 In the end, I just changed the arg in `getTickProps` :smile:

### CODE REVIEW

File: src/h5web/visualizations/shared/models.ts [hunk 1]
```diff
@@ -23,35 +23,20 @@ export type Size = { width: number; height: number };
 
 export type Domain = [number, number];
 
-export interface IndexAxisConfig {
-  indexDomain: Domain;
-  showGrid?: boolean;
-  scaleType?: never; // invalid
-}
```
Comment (loichuder) [inline src/h5web/visualizations/shared/models.ts:30]: One could argue that it is now possible to have integer ticks with a scale that is not linear (and therefore not representative of an IndexAxis).

I think it is fine ! It adds more flexiblity and if the caller wants an index axis, it is its responsibility to set it properly.

File: src/h5web/visualizations/shared/AxisSystem.tsx [hunk 2]
```diff
@@ -62,16 +68,28 @@ function AxisSystem(props: Props): JSX.Element {
...
+  const xTicksScale = SCALE_FUNCTIONS[
+    abscissaConfig.scaleType || ScaleType.Linear
+  ]();
   xTicksScale.domain(visibleDomains[0]);
   xTicksScale.range([0, width]);
 
-  const yTicksScale = ordinateInfo.scaleFn();
+  const yTicksScale = SCALE_FUNCTIONS[
+    ordinateConfig.scaleType || ScaleType.Linear
+  ]();
```
Comment (loichuder) [inline src/h5web/visualizations/shared/AxisSystem.tsx:79]: It is the only strong drawback that I encountered: as `scaleType` can be undefined in `AxisConfig`, I have to supply a fallback it here.

In other places (`Tooltip`, `LineVis`...), the fallback is handled in `getAxisScale`.

Comment (loichuder) [reply src/h5web/visualizations/shared/AxisSystem.tsx]: `scaleType` is now always in `AxisConfig`

File: src/h5web/visualizations/shared/models.ts (hunk 1 above)
Comment (loichuder) [reply src/h5web/visualizations/shared/models.ts:30]: Actually, this conflits with the heatmap: the heatmap cannot accept X/Y axis that are not linear. So I will review this.

File: src/h5web/visualizations/shared/AxisSystemProvider.tsx [hunk 3]
```diff
@@ -1,51 +1,34 @@
...
 
-  const abscissaInfo = getAxisInfo(abscissaConfig);
-  const ordinateInfo = getAxisInfo(ordinateConfig);
-
   return (
-    <AxisSystemContext.Provider value={{ abscissaInfo, ordinateInfo }}>
+    <AxisSystemContext.Provider
+      value={{
+        abscissaConfig: { ...abscissaConfig, scaleType: ScaleType.Linear },
+        ordinateConfig: { ...ordinateConfig, scaleType: ScaleType.Linear },
```
Comment (axelboc) [inline src/h5web/visualizations/shared/AxisSystemProvider.tsx:29]: Hmmm, you're always overriding it like this. You need to destructure the config after the defaults. Perhaps add a `DEFAULT_AXIS_CONFIG` at the top of the file as well, with `scaleType` and perhaps also `isIndexAxis: false` for clarity?

File: src/h5web/visualizations/shared/models.ts [hunk 4]
```diff
@@ -23,19 +23,24 @@ export type Size = { width: number; height: number };
 
 export type Domain = [number, number];
 
-export interface IndexAxisConfig {
-  indexDomain: Domain;
+export type IndexAxisConfig = {
+  domain: Domain;
   showGrid?: boolean;
-  scaleType?: never; // invalid
```
Comment (axelboc) [inline src/h5web/visualizations/shared/models.ts:29]: you want to keep this here (TS is too permissive by default)

File: src/h5web/visualizations/shared/models.ts [hunk 5]
```diff
@@ -23,19 +23,24 @@ export type Size = { width: number; height: number };
...
   scaleType?: ScaleType;
-}
+};
 
-export type AxisConfig = IndexAxisConfig | DataAxisConfig;
+export type AxisConfig = {
+  domain: Domain;
+  showGrid?: boolean;
+  scaleType: ScaleType;
+  isIndexAxis?: boolean;
```
Comment (axelboc) [inline src/h5web/visualizations/shared/models.ts:42]: You could make this required and add `isIndexAxis?: false` in `DataAxisConfig.

Comment (axelboc) [reply src/h5web/visualizations/shared/models.ts:42]: It may also make it easier to add `scaleFn` in this interface, and compute it in `AxisSystemProvider`, before injecting the final config.

**Review by axelboc — CHANGES_REQUESTED** [review]

**Review by axelboc — APPROVED** [review]
😄 better than nothing
